core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -103,27 +104,28 @@ class Queue[A] private (capacity: Int, ref: Ref[State[A]]) {
    * at capacity. If any fibers are interrupted, returns true, otherwise,
    * returns  false.
    */
-  final def interruptOffer[E](t: Throwable): IO[E, Boolean] =
-    IO.flatten(ref.modifyFold[E, IO[E, Boolean]] {
+  final def interruptOffer(t: Throwable): IO[Nothing, Boolean] =
+    IO.flatten(ref.modifyFold[Nothing, IO[Nothing, Boolean]] {
       case Surplus(_, putters) if putters.nonEmpty =>
-        val forked: IO[E, Fiber[E, List[Boolean]]] = IO.forkAll(putters.toList.map(_._2.interrupt[E](t)))
+        val forked: IO[Nothing, Fiber[Nothing, List[Boolean]]] =
+          IO.forkAll[Nothing, Boolean, List](putters.toList.map(_._2.interrupt(t)))
         (forked.flatMap(_.join).map(_.forall(identity)), Deficit(IQueue.empty[Promise[_, A]]))
       case s =>
         (IO.now(false), s)
     })
 
-  private final def removePutter(putter: Promise[_, Unit]): IO[Void, Unit] =
+  private final def removePutter(putter: Promise[_, Unit]): IO[Nothing, Unit] =
     ref
-      .modify[Void] {
+      .modify[Nothing] {
         case Surplus(values, putters) =>
           Surplus(values, putters.filterNot(_._2 == putter))
         case d => d
       }
       .toUnit
 
-  private final def removeTaker(taker: Promise[_, A]): IO[Void, Unit] =
+  private final def removeTaker(taker: Promise[_, A]): IO[Nothing, Unit] =
```

Comment:
We can remove all the existential types in this file (like `Promise[_, A]`) and change them to `Promise[Nothing, A]`, because we never fail the promise so we know they can't have errors.